tools/spectral/ipa/rulesets/functions/eachEnumValueMustBeUpperSnakeCase.js

```diff
@@ -0,0 +1,29 @@
+import { hasException } from './utils/exceptions.js';
+import { getSchemaPath, resolveObject } from './utils/componentUtils.js';
+import { casing } from '@stoplight/spectral-functions';
+
+const RULE_NAME = 'xgen-IPA-123-enum-values-must-be-upper-snake-case';
+const ERROR_MESSAGE = 'enum value must be UPPER_SNAKE_CASE.';
+
+export default (input, _, { path, documentInventory }) => {
+  const oas = documentInventory.resolved;
+  const schemaPath = getSchemaPath(path);
+  const schemaObject = resolveObject(oas, schemaPath);
```

Comment:
```suggestion
  const schemaObject = resolveObject(oas, path);
```
If the `getSchemaPath` is only used for the `resolveObject` function, perhaps `getSchemaPath` can be a private function in utils called internally by `resolveObject`

Reply:
The `resolveObject` function can be used for many custom functions in the future to resolve the object for the given path, it is not specific to get the schema object